Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -581,6 +582,12 @@ void WiiSocket::Update(bool read, bool write, bool except)
 
           // Not a string, Windows requires a const char* for sendto
           const char* data = (const char*)memory.GetPointer(BufferIn);
+          const std::optional<std::string> patch = WC24PatchEngine::GetNetworkPatchByPayload(data);
+          if (patch)
+          {
+            data = patch->c_str();
+            BufferInSize = static_cast<u32>(patch->size()) - 2;
```

Comment:
Upon reviewing payload sizes and how they look, all Nintendo Channel payloads were `data_size - 2`. This is because the literal `FR` was at the end of the string, not sure what it is for but it should not be sent through the socket.

Reply:
Oh, duh, yeah that's not a nullterminated string so we're treating it incorrectly. That explains your corruption. Please change it to

```
          const std::optional<std::string> patch = WC24PatchEngine::GetNetworkPatchByPayload(std::string_view(data, data + BufferInSize));
          if (patch)
          {
            data = patch->c_str();
            BufferInSize = static_cast<u32>(patch->size()) - 2;
          }
```

and see if it's correct then.

Reply:
Works like a charm.